### PR TITLE
Minor refactor, README changes and LICENSE change

### PR DESCRIPTION
**Pull request by chocol4te**
- Minor refactor, mostly readability changes
- Add Tokei count badge, link to contributing guidelines and link to license in README.md
- Change license to 0BSD

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,116 +1,5 @@
-CC0 1.0 Universal
+Copyright (C) 2018-2019 by Jonas Schievink
```
Comment (jonas-schievink) [inline LICENSE]: "by" is probably not a good idea in a copyright notice. See https://ben.balter.com/2015/06/03/copyright-notices-for-websites-and-open-source-projects/ for how this should look in open-source projects.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,20 @@
-# WIP Rust BLE stack
+# rubble
+
+> A WIP Rust BLE stack
 
 [![Build Status](https://travis-ci.org/jonas-schievink/rubble.svg?branch=master)](https://travis-ci.org/jonas-schievink/rubble)
+[![Line Count](https://tokei.rs/b1/github/jonas-schievink/rubble)](https://github.com/XAMPPRocky/tokei)
```
Comment (jonas-schievink) [inline README.md]: I'm not a big fan of bragging about line counts. They don't really convey much information, and might even keep away new contributors when the numbers look too large.

Comment (chocol4te) [reply README.md]: Good point, removed.

File: LICENSE (hunk 1 above)
Comment (chocol4te) [reply LICENSE]: Interesting read, done

File: README.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
-# WIP Rust BLE stack
+# rubble
+
+> A Bluetooth Low-Energy stack for embedded Rust
```
Comment (jonas-schievink) [inline README.md]: I avoided the spelled-out wording due to the copyright on the word "Bluetooth", can you change that back? Also, this isn't a quote, so the `>` should also be removed.

File: src/ble/crc.rs [hunk 4]
```diff
@@ -4,14 +4,14 @@ use super::link::CRC_POLY;
...
+    let polynomial = CRC_POLY & 0x00FF_FFFF;
     let mut crc = preset;
 
     for byte in data {
-        crc = crc ^ (*byte as u32) << 16;
+        crc ^= u32::from(*byte) << 16;
 
         for _ in 0..8 {
-            let msb = crc & 0x800000 != 0;
+            let msb = crc & 0x0080_0000 != 0;
```
Comment (jonas-schievink) [inline src/ble/crc.rs]: This was more readable before this change

File: src/ble/link/advertising.rs [hunk 5]
```diff
@@ -29,15 +29,15 @@ use {
...
+pub const CRC_PRESET: u32 = 0x00555555;
 
 /// Max. advertising PDU payload size in Bytes.
 ///
 /// Note that data channel PDUs can carry much larger payloads.
 pub const MAX_PAYLOAD_SIZE: usize = 37;
 
 /// Access Address to use for all advertising channel packets.
-pub const ACCESS_ADDRESS: u32 = 0x8E89BED6;
+pub const ACCESS_ADDRESS: u32 = 0x8E89_BED6;
```
Comment (jonas-schievink) [inline src/ble/link/advertising.rs]: The `_` doesn't seem helpful here

File: src/ble/crc.rs [hunk 6]
```diff
@@ -4,14 +4,14 @@ use super::link::CRC_POLY;
 ///
 /// The result has 24 bits stored in the lower 24 bits of a `u32`.
 pub fn ble_crc24(data: &[u8], preset: u32) -> u32 {
-    let polynomial = CRC_POLY & 0xFFFFFF;
+    let polynomial = CRC_POLY & 0x00FF_FFFF;
```
Comment (jonas-schievink) [inline src/ble/crc.rs]: The `_` doesn't really help here, it's a 24-bit value

File: src/ble/crc.rs [hunk 7]
```diff
@@ -20,5 +20,5 @@ pub fn ble_crc24(data: &[u8], preset: u32) -> u32 {
         }
     }
 
-    crc & 0xFFFFFF
+    crc & 0x00FF_FFFF
```
Comment (jonas-schievink) [inline src/ble/crc.rs]: The `_` doesn't really help here, it's a 24-bit value

File: src/ble/link/advertising.rs [hunk 8]
```diff
@@ -694,13 +694,13 @@ impl fmt::Debug for PduBuf {
 #[derive(Copy, Clone)]
 pub struct Header(u16);
 
-const TXADD_MASK: u16 = 0b00000000_01000000;
-const RXADD_MASK: u16 = 0b00000000_10000000;
+const TXADD_MASK: u16 = 0b0000000001000000;
+const RXADD_MASK: u16 = 0b0000000010000000;
```
Comment (jonas-schievink) [inline src/ble/link/advertising.rs]: These underscores were there for a reason though - this is way harder to read now

File: src/ble/link/data.rs [hunk 9]
```diff
@@ -152,16 +152,16 @@ impl Header {
...
-        let bit = self.0 & 0b1_0000;
+        let bit = self.0 & 0b10000;
         bit != 0
     }
 
     /// Sets the value of the `MD` field.
     pub fn set_md(&mut self, md: bool) {
         if md {
-            self.0 |= 0b1_0000;
+            self.0 |= 0b0000;
```
Comment (jonas-schievink) [inline src/ble/link/data.rs]: This is incorrect

File: src/ble/phy.rs [hunk 10]
```diff
@@ -19,7 +19,7 @@ fn rf_channel_freq(rf_channel: u8) -> u16 {
 /// Returns the data whitening IV for a channel index (not RF channel).
 fn whitening_iv(channel_idx: u8) -> u8 {
     debug_assert!(channel_idx <= 39);
-    0b01000000 | channel_idx
+    0b0100000 | channel_idx
```
Comment (jonas-schievink) [inline src/ble/phy.rs]: This also introduces a bug

File: src/radio.rs [hunk 11]
```diff
@@ -103,7 +103,7 @@ impl BleRadio {
             });
             radio
                 .crcpoly
-                .write(|w| w.crcpoly().bits(CRC_POLY & 0xFFFFFF));
+                .write(|w| w.crcpoly().bits(CRC_POLY & 0x00FF_FFFF));
```
Comment (jonas-schievink) [inline src/radio.rs]: `_` doesn't seem useful here

File: src/radio.rs [hunk 12]
```diff
@@ -345,7 +345,7 @@ impl BleRadio {
                 .write(|w| w.datawhiteiv().bits(channel.whitening_iv()));
             self.radio
                 .crcinit
-                .write(|w| w.crcinit().bits(crc_init & 0x00ffffff));
+                .write(|w| w.crcinit().bits(crc_init & 0x00FF_FFFF));
```
Comment (jonas-schievink) [inline src/radio.rs]: `_` doesn't seem useful here

File: src/ble/crc.rs [hunk 13]
```diff
@@ -4,14 +4,14 @@ use super::link::CRC_POLY;
...
+    let polynomial = CRC_POLY & 0x00FFFFFF;
     let mut crc = preset;
 
     for byte in data {
-        crc = crc ^ (*byte as u32) << 16;
+        crc ^= u32::from(*byte) << 16;
 
         for _ in 0..8 {
-            let msb = crc & 0x800000 != 0;
+            let msb = crc & 0x0080000 != 0;
```
Comment (jonas-schievink) [inline src/ble/crc.rs]: This change to the constant doesn't seem right

**Pull request merged**